### ISSUE TITLE
Firmware difference 

### ISSUE THREAD

**Issue by Stan2Be** [atc1441/ATC_RF03_Ring#7]
Hi!

Whats exactly the difference between the original OTA Firmware and the one you posted later:

R02_3.00.06_FasterRawValuesMOD.bin

**Comment by atc1441** [comment]
This enables faster reading of RAW sensor values. 
See the https://atc1441.github.io/ATC_RF03_Writer.html tool and click on "Enable Animation" before and after to see the difference

**Comment by Stan2Be** [comment]
thank you for your fast answer! 

--- Ursprüngliche Nachricht --- 
Von: atc1441 ***@***.***>
Datum: 06.07.2024 18:11
An: atc1441/ATC_RF03_Ring ***@***.***>
Betreff: Re: [atc1441/ATC_RF03_Ring] Firmware difference  (Issue #7)

> 
>This enables faster reading of RAW sensor values.
>See the  https://atc1441.github.io/ATC_RF03_Writer.html  tool and click 
>on "Enable Animation" before and after to see the difference 
>—
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>You are receiving this because you authored the thread.  Message ID:  
><atc1441/ATC_RF03_Ring/issues/7/2211808653  @  github  .  com>    
>***@***.***": ***@***.***": 
>"EmailMessage","potentialAction": ***@***.***": "ViewAction","target": 
>ithub.com/atc1441/ATC_RF03_Ring/issues/7#issuecomment-2211808653","url": 
>thub.com/atc1441/ATC_RF03_Ring/issues/7#issuecomment-2211808653","name": 
>"View Issue"},"description": "View this Issue on GitHub","publisher": 
>***@***.***": "Organization","name": "GitHub","url": "https://github.com"}}] 

________________________________________________________
Your E-Mail. Your Cloud. Your Office. eclipso Mail Europe. https://www.eclipso.de

**Comment by Stan2Be** [comment]
Does this firmware fits to the R03 or R06 ring? Ir only for the R02?

**Comment by otonieru** [comment]
> Does this firmware fits to the R03 or R06 ring? Ir only for the R02?

so i bite the bullet and tried to flash the firmware to my own R06. Happy to tell that it works just fine, aside that the device name now change to R02. But i think it can be modified in the .bin itself. Might try to change it myself later

cc : @atc1441

**Comment by Stan2Be** [comment]
I tried to flash it to the R03 it wrote wrong CRC... so it doest work. But im also getting R06 soon so would you be so kind and upload the file for the R06 firmware too? I cant modify it myself... thank you

**Comment by otonieru** [comment]
> I tried to flash it to the R03 it wrote wrong CRC... so it doest work. But im also getting R06 soon so would you be so kind and upload the file for the R06 firmware too? I cant modify it myself... thank you

I use the MOD firmware from here : https://github.com/atc1441/ATC_RF03_Ring/blob/main/R02_3.00.06_FasterRawValuesMOD.bin and it actually works

**Comment by atc1441** [comment]
@Stan2Be  you can just flash it to R03 and R06 even while the Bluetooth name changes then to R03 for both its not an issue as the hardware is the same

**Comment by Stan2Be** [comment]
I tried but it wrote CRC error and stopped

**Comment by Stan2Be** [comment]
It worked with the modded firmware but now renamed the ring to R02 instead of R03…. How can I change it back?

**Comment by Stan2Be** [comment]
Is there a way to rename the ring ahsin? Because i have another R02 and now both comes as R02...

**Comment by adiif1** [comment]
@atc1441 Hi!, if possible to change read value in apk qring from 5 min to 1?  I use your mod software

**Comment by Stan2Be** [comment]
i cant do such thing alone... im a newbie to this things :(

--- Ursprüngliche Nachricht --- 
Von: adiif1 ***@***.***>
Datum: 10.08.2024 09:13
An: atc1441/ATC_RF03_Ring ***@***.***>
Betreff: Re: [atc1441/ATC_RF03_Ring] Firmware difference  (Issue #7)

> 
> @atc1441  Hi!, if possible to change read value in apk qring from 5 min 
>to 1?  I use your mod software 
>—
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>You are receiving this because you were mentioned.  Message ID:  
><atc1441/ATC_RF03_Ring/issues/7/2279911059  @  github  .  com>    
>***@***.***": ***@***.***": 
>"EmailMessage","potentialAction": ***@***.***": "ViewAction","target": 
>ithub.com/atc1441/ATC_RF03_Ring/issues/7#issuecomment-2279911059","url": 
>thub.com/atc1441/ATC_RF03_Ring/issues/7#issuecomment-2279911059","name": 
>"View Issue"},"description": "View this Issue on GitHub","publisher": 
>***@***.***": "Organization","name": "GitHub","url": "https://github.com"}}] 

________________________________________________________
Your E-Mail. Your Cloud. Your Office. eclipso Mail Europe. https://www.eclipso.de

**Comment by devnoname120** [comment]
@atc1441 How hard would it be to increase the frequency of automatic HR/SPO₂ measures beyond the 5 minute/1 hour treshold?

**Comment by Stan2Be** [comment]
i just don't know how to make it... i just found your Guide and Thats how i Flash it but i can't do more... can you help me?

--- Ursprüngliche Nachricht --- 
Von: Paul ***@***.***>
Datum: 16.08.2024 18:04
An: atc1441/ATC_RF03_Ring ***@***.***>
Betreff: Re: [atc1441/ATC_RF03_Ring] Firmware difference  (Issue #7)

> 
> @atc1441  How hard would it be to increase the frequency of automatic 
>HR/SPO₂ measures beyond the 5 minute/1 hour treshold? 
>—
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>You are receiving this because you were mentioned.  Message ID:  
><atc1441/ATC_RF03_Ring/issues/7/2293770257  @  github  .  com>    
>***@***.***": ***@***.***": 
>"EmailMessage","potentialAction": ***@***.***": "ViewAction","target": 
>ithub.com/atc1441/ATC_RF03_Ring/issues/7#issuecomment-2293770257","url": 
>thub.com/atc1441/ATC_RF03_Ring/issues/7#issuecomment-2293770257","name": 
>"View Issue"},"description": "View this Issue on GitHub","publisher": 
>***@***.***": "Organization","name": "GitHub","url": "https://github.com"}}] 

________________________________________________________
Your E-Mail. Your Cloud. Your Office. eclipso Mail Europe. https://www.eclipso.de

**Comment by adiif1** [comment]
> i just don't know how to make it... i just found your Guide and Thats how i Flash it but i can't do more... can you help me?
> […](#)
> --- Ursprüngliche Nachricht --- Von: Paul ***@***.***> Datum: 16.08.2024 18:04 An: atc1441/ATC_RF03_Ring ***@***.***> Betreff: Re: [atc1441/ATC_RF03_Ring] Firmware difference (Issue #7)
> @atc1441 How hard would it be to increase the frequency of automatic HR/SPO₂ measures beyond the 5 minute/1 hour treshold? — Reply to this email directly, view it on GitHub , or unsubscribe . You are receiving this because you were mentioned. Message ID: <atc1441/ATC_RF03_Ring/issues/7/2293770257 @ github . com> ***@***.***": ***@***.***": "EmailMessage","potentialAction": ***@***.***": "ViewAction","target": ithub.com/atc1441/ATC_RF03_Ring/issues/7#issuecomment-2293770257","url": thub.com/atc1441/ATC_RF03_Ring/issues/7#issuecomment-2293770257","name": "View Issue"},"description": "View this Issue on GitHub","publisher": ***@***.***": "Organization","name": "GitHub","url": "https://github.com"}}]
> ________________________________________________________ Your E-Mail. Your Cloud. Your Office. eclipso Mail Europe. https://www.eclipso.de

buddy but read from beginning to end.. see who we mark as a clue - the author...

**Comment by Andre-Lehrmann** [comment]
Hi, i flashed a R06 Ring in 12 and in 11 with [R02_3.00.06_FasterRawValuesMOD.bin] (https://github.com/atc1441/ATC_RF03_Ring/blob/main/R02_3.00.06_FasterRawValuesMOD.bin) and both rings are working. 

Is it possible to change the firmware so that the ring measures every 6 minutes?

According to adverts, the ring is supposed to be able to store 7 days of data. That is 48 measurements per day times 7 = 336 measuring points.
If measurements are now taken every 6 minutes, then the time until the mobile phone has to read out the data at the latest would be: 6 * 336 =2016 minutes = approx. 33 hours. This would get you through the day without any problems and give you more accurate results.

**Comment by Stan2Be** [comment]
I know i can flash it i just don’t want the name of the ring to be changed to R02 like it happened with my R03 ring…

**Comment by atc1441** [comment]
> I know i can flash it i just don’t want the name of the ring to be changed to R02 like it happened with my R03 ring…

The name does not matter actually. But your decision of course :)

**Comment by Stan2Be** [comment]
True! What’s why u asked if there is a possibility to change the name in the firmware to R06… but @atc1441  didn’t answered…

**Comment by atc1441** [comment]
Yes it can be changed as shown in the Video :)
https://youtu.be/IOMqtFrNpTI

**Comment by Stan2Be** [comment]
I just watched the video again but I’m totally newbie into this and I didn’t understood how to change the name of the ring when I flash the software… 😢

**Comment by Andre-Lehrmann** [comment]
I know this is not your job @atc1441 , but could you please give me a tip on which tool I can use to change the firmware? 

I don't mean HeXd, as I can already change values there, 
but that would be rather random. Is there a tool that translates the firmware into any programming language (except assembler!), something like C# or another higher programming language? 


@StanBe - you have to find the right place where the firmware name is written. Change this value with the Hex tool and enter a new checksum at the top. Then you have patched it yourself.

**Comment by Stan2Be** [comment]
That means I have to download something on my computer so I can see all this things? I used only the flash tool until now…

**Comment by Andre-Lehrmann** [comment]
Yes, you need a Hex Editor, use the recommend in the video. And the firmware as a file.

**Comment by Stan2Be** [comment]
Thank you I will give it a try

**Comment by atc1441** [comment]
> I know this is not your job @atc1441 , but could you please give me a tip on which tool I can use to change the firmware?
> 
> I don't mean HeXd, as I can already change values there, but that would be rather random. Is there a tool that translates the firmware into any programming language (except assembler!), something like C# or another higher programming language?
> 
> @stanbe - you have to find the right place where the firmware name is written. Change this value with the Hex tool and enter a new checksum at the top. Then you have patched it yourself.

There is no tool like that.

You can use Ghidra to look into the firmware but edits need to be on hex level

**Comment by Stan2Be** [comment]
@atc1441 i tried with this software but it didn’t worked :( is it a lot of work if you change it for me to R06v? Please 🙏

**Comment by Andre-Lehrmann** [comment]
> Hi, i flashed a R06 Ring in 12 and in 11 with [R02_3.00.06_FasterRawValuesMOD.bin] (https://github.com/atc1441/ATC_RF03_Ring/blob/main/R02_3.00.06_FasterRawValuesMOD.bin) and both rings are working.
> 
> Is it possible to change the firmware so that the ring measures every 6 minutes?
> 
> According to adverts, the ring is supposed to be able to store 7 days of data. That is 48 measurements per day times 7 = 336 measuring points. If measurements are now taken every 6 minutes, then the time until the mobile phone has to read out the data at the latest would be: 6 * 336 =2016 minutes = approx. 33 hours. This would get you through the day without any problems and give you more accurate results.

UPS, i found a option to set every 5 minutes in the app. Sorry, my fault.

**Comment by balaclava-guy** [comment]
@Stan2Be how did you get past the CRC error?

**Comment by AnalyticETH** [comment]
Version 3.00.17 has been released. Can the modified firmware be updated for this? Does anyone have a dump of this latest firmware? Thanks
